pkg/controller/pod/pod_controller.go

```diff
@@ -179,6 +179,13 @@ func (r *ReconcilePod) isForNsxNodeAgentPod(request reconcile.Request) bool {
 	if request.Namespace == operatortypes.NsxNamespace && strings.Contains(
 		request.Name, operatortypes.NsxNodeAgentDsName) &&
 		request.Name != operatortypes.NsxNodeAgentDsName {
+		// Check if the pod exists to prevent that a stale pod is in a endless periodic resync
```

Comment:
If nsx-node-agent pod is deleted, could you set node status to False in this case?

Reply:
When the operator checks network-unavailable status, it won't update the status if it fails to get the nsx-node-agent pod because this case is usually that the nsx-node-agent is updated, i.e. old pod is deleted and new pod is created, it's safe to ignore the event from old pod.

Reply:
reverted this change in new patch set